trace/sql/sql.go

```diff
@@ -39,79 +39,77 @@ type DSNInfo struct {
 func (c *Conn) BeginTx(ctx context.Context, opts *sql.TxOptions) (*Tx, error) {
 	sp, _ := c.startSpan(ctx, "conn.BeginTx", "")
 	tx, err := c.conn.BeginTx(ctx, opts)
+	defer trace.SpanComplete(sp, err)
```

Comment:
No need to defer here.
```suggestion
	trace.SpanComplete(sp, err)
```